File: quick_sort.c
void quick_sort(int arr[], int low, int high) {
    if (low < high) {
        int pivot = arr[low];
        int left = low;
        int right = high;

        while (left < right) {
            while (arr[right] >= pivot && left < right) {
                right--;
            }
            arr[left] = arr[right];
            while (arr[left] <= pivot && left < right) {
                left++;
            }
            arr[right] = arr[left];
        }
        arr[left] = pivot;

        quick_sort(arr, low, left - 1);
        quick_sort(arr, left + 1, high);
    }
}

void  main() {
    int arr[10];
    for (int i = 0; i < 10; i++) {
        arr[i] = 10 - i;
    }

    quick_sort(arr, 0, 9);
}

File: quick_sort.rs
fn quick_sort(arr: &mut [i32], low: usize, high: usize) {
    if low < high {
        let pivot = arr[low];
        let mut left = low;
        let mut right = high;

        while left < right {
            while arr[right] >= pivot && left < right {
                right -= 1;
            }
            arr[left] = arr[right];
            while arr[left] <= pivot && left < right {
                left += 1;
            }
            arr[right] = arr[left];
        }
        arr[left] = pivot;

        quick_sort(arr, low, left - 1);
        quick_sort(arr, left + 1, high);
    }
}

fn main() {
    let mut arr = [0; 10];
    for i in 0..10 {
        arr[i] = 10 - i as i32;
    }

    quick_sort(&mut arr, 0, 9);
}
